From 32982eff1abe7653f96f746cc15c0d6d1bbbb249 Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Tue, 14 Mar 2023 14:29:54 -0600
Subject: [PATCH] kcms/nightcolor: fix timing strings with narrow window widths

These labels lived in an item that had `anchors.centerIn: parent` set on
it, which makes it have no maximum width and it can overflow. Clearly it
was the intent to avoid that, because all the child items had their
properties set up correctly for it.

To fix this, instead anchor to the parent's size and center the
contents, and then make the parent fill the available width.

Resolves https://invent.kde.org/plasma/plasma-mobile/-/issues/244


(cherry picked from commit 16a243c88b6f5e36ec4d3f9108f2e967b4e97ad8)
---
 kcms/nightcolor/package/contents/ui/main.qml | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/kcms/nightcolor/package/contents/ui/main.qml b/kcms/nightcolor/package/contents/ui/main.qml
index 60605b3180..c37cc0ea9a 100644
--- a/kcms/nightcolor/package/contents/ui/main.qml
+++ b/kcms/nightcolor/package/contents/ui/main.qml
@@ -406,7 +406,7 @@ KCM.SimpleKCM {
             visible: kcm.nightColorSettings.mode === NightColorMode.Automatic || kcm.nightColorSettings.mode === NightColorMode.Location
                 && kcm.nightColorSettings.active
             Layout.topMargin: Kirigami.Units.largeSpacing * 4
-            Layout.alignment: Qt.AlignHCenter
+            Layout.fillWidth: true
 
             Kirigami.LoadingPlaceholder {
                 visible: kcm.nightColorSettings.active && kcm.nightColorSettings.mode === NightColorMode.Automatic && (!locator || !root.doneLocating)
@@ -416,9 +416,13 @@ KCM.SimpleKCM {
 
             TimingsView {
                 id: timings
+                anchors {
+                    top: parent.top
+                    left: parent.left
+                    right: parent.right
+                }
                 visible: kcm.nightColorSettings.mode === NightColorMode.Location ||
                     (kcm.nightColorSettings.mode === NightColorMode.Automatic && root.doneLocating) && kcm.nightColorSettings.active
-                anchors.centerIn: parent
                 enabled: kcm.nightColorSettings.active
                 latitude: kcm.nightColorSettings.mode === NightColorMode.Automatic
                     && (locator !== undefined) ? locator.latitude : kcm.nightColorSettings.latitudeFixed
-- 
GitLab

